src/content/app/tools/vep/components/vep-submission-header/VepSubmissionHeader.tsx

```diff
@@ -42,47 +47,101 @@ type Props = {
+                Submission
+              </span>
+              {submission.id}
+            </>
+          )}
+        </div>
+        <div className={styles.rerun}>
+          <TextButton onClick={noop}>Edit/rerun</TextButton>
+        </div>
+        <div className={styles.submissionDate}>
+          {submissionTime} <span className={styles.timezone}>GMT</span>
+        </div>
+        <ControlButtons
+          {...props}
+          isDeleting={isDeleting}
+          onDelete={toggleDeletionConfirmation}
+        />
       </div>
-      <div className={styles.rerun}>
-        <TextButton onClick={noop}>Edit/rerun</TextButton>
-      </div>
-      <div className={styles.submissionDate}>
-        {submissionTime} <span className={styles.timezone}>GMT</span>
-      </div>
-      <ControlButtons {...props} />
-    </div>
+      {isDeleting && (
+        <div className={styles.deletionConfirmationContainer}>
+          <DeletionConfirmation
+            {...props}
+            onCancel={toggleDeletionConfirmation}
+          />
+        </div>
+      )}
+    </>
   );
 };
 
-const ControlButtons = (props: Props) => {
-  const { submission } = props;
-  const dispatch = useAppDispatch();
+const hasServerSideSubmissionId = (submission: { status: string }) => {
+  // Only submissions with a server-side status will have an id assigned by the server
+  // (temporary client-side ids aren't helpful for users; so there's no point in displaying those)
+  return (serverSideSubmissionStatuses as readonly string[]).includes(
+    submission.status
+  );
+};
+
+const ControlButtons = (
+  props: Props & {
+    isDeleting: boolean;
+    onDelete: () => void;
+  }
+) => {
+  const { submission, isDeleting, onDelete } = props;
 
   const canGetResults = submission.status === 'SUCCEEDED';
 
+  return (
+    <div className={styles.controls}>
+      <DeleteButton onClick={onDelete} disabled={isDeleting} />
+      <DownloadButton onClick={noop} disabled={isDeleting || !canGetResults} />
+      <ButtonLink isDisabled={isDeleting || !canGetResults} to="/">
+        Results
+      </ButtonLink>
+    </div>
+  );
+};
+
+const DeletionConfirmation = (
+  props: Props & {
+    onCancel: () => void;
+  }
+) => {
+  const { submission, onCancel } = props;
+  const dispatch = useAppDispatch();
+
   const onDelete = () => {
     dispatch(deleteSubmission({ submissionId: submission.id }));
   };
 
   return (
-    <div className={styles.controls}>
-      <DeleteButton onClick={onDelete} />
-      <DownloadButton onClick={noop} disabled={!canGetResults} />
-      <ButtonLink isDisabled={!canGetResults} to="/">
-        Results
-      </ButtonLink>
+    <div className={styles.deletionConfirmation}>
+      <span className={styles.deletionConfirmationMessage}>
+        Delete this submisison?
+      </span>
+      <PrimaryButton onClick={onDelete}>Delete</PrimaryButton>
+      <TextButton onClick={onCancel}>Do not delete</TextButton>
```

Comment:
Shouldn't this be the `SecondaryButton`?

Reply:
This is what the `SecondaryButton` looks like in our storybook: https://ensembl.github.io/ensembl-client/storybook/index.html?path=/story/components-shared-components-button--default-secondary-button. I can't remember how often we've used it; probably very infrequently.

TextButton is very different. It doesn't have an appearance of a button. It looks like blue clickable text. We still have a lot of places in our codebase where we used a `span` or a `div` with a click handler. This was totally my fault. Clickable text that is not a link should be represented with an html `button` element. So several months ago, I created the `TextButton` component specifically for that purpose.